### PR TITLE
cows_bulls_function created.

### PR DESCRIPTION
**Pull request by v1020**

### CODE REVIEW

File: hw4/Ex4/server/cows_bulls.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * @brief: The function computes cows_bulls game based on the number and a guess.  
+ * The result is a string in a format of C(num)B(num)
+ * 
+ * @param[in] 3 strings: player_num, guess and result
+ * @param[out] result string will be filled
+ * @param[out] game result
+ * @return error_e
+ */
+error_e cows_bulls(char player_num[], char guess[], char result[])
```
Comment (LevyMatan) [inline hw4/Ex4/server/cows_bulls.c:16]: Need to add input integrity validation:
You are reading and writing from array, must validate the pointer are not NULLs.

Comment (LevyMatan) [inline hw4/Ex4/server/cows_bulls.c:16]: Need to think on a better name. Silly but important.
maybe count_bulls_and_cows

**Review by LevyMatan — CHANGES_REQUESTED** [review]

File: hw4/Ex4/server/cows_bulls.c (hunk 1 above)
Comment (v1020) [reply hw4/Ex4/server/cows_bulls.c]: fixed

Comment (v1020) [reply hw4/Ex4/server/cows_bulls.c]: name changed

**Review by LevyMatan — APPROVED** [review]

**Pull request merged**